### DISCUSSION TITLE
use brython in nicegui?

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2244] [Q&A]
### Question

I try to embed a python interpreter in WEB server. [Brython (Browser Python)](https://github.com/brython-dev/brython) seems to be an easy one, although it is not as powerful as https://github.com/pyscript/pyscript or https://github.com/skulpt/skulpt

The original HTML is so straightforward
```html
<html>

<head>
<meta charset="utf-8">
<script type="text/javascript" src="brython.js"></script>
<script type="text/javascript" src="brython_stdlib.js"></script>
</head>

<body onload="brython(1)">
<script type="text/python">
from browser import document

document <= "Hello"
</script>
</body>

</html>
```

but the first question is, how to translate `<body onload="brython(1)">` in to NiceGUI? Thanks

**Comment by falkoschindler** [comment]
See #1904 for a related discussion about integrating Brython with NiceGUI. I'm still sceptical if this makes sense, but I'd be happy to be proven wrong.

Regarding the "onload" event handler:
```py
ui.add_head_html('<script>window.onload = (e) => console.log("onload", e);</script>')
```
This JavaScript snippet lets you register any function to be called when the window is loaded.
